src/graphql/root/mutation/user-update-username.ts

```diff
@@ -35,7 +34,9 @@ const UserUpdateUsernameMutation = GT.Field({
     if (result instanceof Error) {
       return {
         errors: [mapAndParseErrorForGqlResponse(result)],
-        ...(result instanceof UsernameIsImmutableError ? { user: domainUser } : {}),
+
+        // FIXME: what is this return for?
+        // ...(result instanceof UsernameIsImmutableError ? { user: domainUser } : {}),
```

Comment:
Commenting this out seems like a breaking change in API behaviour. Need to validate.

Reply:
@samerbuna do you know what this line is for?

At least on the happy path (username is correctly set) commenting this line doesn't change anything

Reply:
It would return the user still in the response if the error type is `UsernameIsImmutableError`. Commenting here means that the `user` selection will always be undefined if `result` is an error.

There may be use-cases that rely on getting that user back even if the username value passed is invalid?

Reply:
Related: https://github.com/GaloyMoney/galoy/pull/1991